cli/internal/terraform/terraform/azure/modules/scale_set/main.tf

```diff
@@ -11,6 +11,19 @@ terraform {
   }
 }
 
+locals {
+  tags = merge(
+    var.tags,
+    { constellation-role = var.role },
```

Comment:
This should perform the transformation `ControlPlane` => `control-plane` and `Worker`=>`worker`